docs/source/kiva_tutorial/tutorial.rst

```diff
@@ -0,0 +1,172 @@
+the arguments specifying the point to draw at represent the lower left corner
+of the resulting text.  Thus, since the origin of our coordinate system is
+currently at the center of the circle, in order for the text to be centered we
+need to draw the text at (-w/2, -h/2).
+
+.. literalinclude:: tutorial.py
+    :lines: 28-51
+
+.. image:: images/step_3.png
+  :width: 600
+  :height: 400
+
+As you may have noticed, most of the code for drawing the Ammeter and the
+Voltmeter was effectively the same. Sometimes it is useful to work with an
+independent path instance as opposed to specifically messing with the current
+path of the graphics context. This brings us to the notion of CompiledPaths,
+which we will now use to draw the resistors. As you can see the path for each
+resistor will be exactly the same. Rather than moving our coordinate system
+around to each location and redrawing the same path at each location, we will
+instead define the path using a :class:CompiledPath and draw at each of the
+various desired locations using the :py:meth:`draw_path_at_points` method. To
+do this we instantiate a :class:`CompiledPath`, and then define our path just
+as we would with the graphics context's current path. The interface uses the
+same :ref:`kiva_path_functions`. Finally, we can simply call
+:py:meth:`draw_path_at_points` passing in the locations we want to draw the
+path, our compiled path, and the drawing mode.
+
+.. literalinclude:: tutorial.py
+    :lines: 70-77
+
+If you run just this code, you will notice that things don't look quite right,
+as the original lines for the wires still show underneath our resistors. We can
+get rid of these by drawing white lines over the relevant portions of the wire
+before we call the above code. Note that this is not the most performant
+approach we could take especially in the context of an interactive application.
+In general we want to strive to touch each pixel as few times as necessary.
+However, in this case to keep the tutorial simple, we can run 
+
+.. literalinclude:: tutorial.py
+    :lines: 54-68
+
+.. image:: images/step_45.png
+  :width: 600
+  :height: 400
+
+
+Now, for the switch. Just like above, we are going to want to "remove" the
+previously drawn wire. Then, for the actual switch, we want to effectively just
+rotate that segment of wire outwards. To do we tranform our coordinates so that
+we can simply draw a line along the x axis of the same length as the gap we
+just created. Thus, we translate to the lower edge of the switch, and rotate
+our axis to our desired angle. The beauty of the context manager then reverts
+our coordinate system back to as it was before.
+
+.. literalinclude:: tutorial.py
+    :lines: 80-98
+
+.. image:: images/step_6.png
+  :width: 600
+  :height: 400
+
+We will leave the drawing of the battery as an exercise for the reader, but the
+full code for the example is available :download:`here <tutorial.py>`.
```

Comment:
Here too, I suggest wrapping the code in a bare .rst file with a single `literalinclude` and then using `:ref:` instead of `:download:` for the linking.